File: reactor/r2dbc-mysql/src/main/java/ua/mai/zyme/r2dbcmysql/config/AppConfig.java
package ua.mai.zyme.r2dbcmysql.config;


import io.r2dbc.spi.ConnectionFactory;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.r2dbc.connection.R2dbcTransactionManager;
import org.springframework.r2dbc.connection.init.ConnectionFactoryInitializer;

@Configuration
public class AppConfig {

    @Bean
    ConnectionFactoryInitializer initializer(@Qualifier("connectionFactory")
                                             ConnectionFactory connectionFactory) {
        ConnectionFactoryInitializer initializer = new ConnectionFactoryInitializer();
        initializer.setConnectionFactory(connectionFactory);
//        ResourceDatabasePopulator resource =
//                new ResourceDatabasePopulator(new ClassPathResource("schema.sql"));
//        initializer.setDatabasePopulator(resource);
        return initializer;
    }


//    @Bean
//    public R2dbcTransactionManager transactionManager(@Qualifier("connectionFactory")
//                                                      ConnectionFactory connectionFactory) {
//        return new R2dbcTransactionManager(connectionFactory);
//    }

}

File: kotlin/kotlin-r2dbc-mysql/kotlin-r2dbc-mysql-be-app/src/main/java/ua/mai/zyme/r2dbcmysql/config/AppConfig.kt
package ua.mai.zyme.r2dbcmysql.config

import io.r2dbc.spi.ConnectionFactory
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.ClassPathResource
import org.springframework.r2dbc.connection.init.ConnectionFactoryInitializer
import org.springframework.r2dbc.connection.init.ResourceDatabasePopulator
import org.springframework.r2dbc.connection.R2dbcTransactionManager

@Configuration
open class AppConfig {

    @Bean
    open fun initializer(@Qualifier("connectionFactory") connectionFactory: ConnectionFactory): ConnectionFactoryInitializer {
        val initializer = ConnectionFactoryInitializer()
        initializer.setConnectionFactory(connectionFactory)
//        val resource = ResourceDatabasePopulator(ClassPathResource("schema.sql"))
//        initializer.setDatabasePopulator(resource)
        return initializer
    }

//    @Bean
//    open fun transactionManager(@Qualifier("connectionFactory") connectionFactory: ConnectionFactory): R2dbcTransactionManager {
//        return R2dbcTransactionManager(connectionFactory)
//    }

}